frontend/public/module/k8s/get-resources.ts

```diff
@@ -136,7 +139,41 @@ export const getResources = () =>
       const safeResources = [];
       const adminResources = [];
 
+      const getModelExtensionMetadata = (
+        extensions: LoadedExtension<ModelMetadata>[],
+        group: string,
+        version?: string,
+        kind?: string,
+      ) => {
+        const groupVersionKindMetadata = extensions.filter(
+          ({ properties }) =>
+            properties?.model?.group === group &&
+            properties?.model?.kind === kind &&
+            properties?.model?.version === version,
+        );
+        if (groupVersionKindMetadata.length > 0) {
+          return Object.assign({}, ...groupVersionKindMetadata.map((e) => e.properties));
+        }
+        const groupKindMetadata = extensions.filter(
+          ({ properties }) =>
+            properties?.model?.group === group && properties?.model?.kind === kind,
+        );
+        if (groupKindMetadata.length > 0) {
+          return Object.assign({}, ...groupKindMetadata.map((e) => e.properties));
+        }
+        const groupMetadata = extensions.filter(
+          ({ properties }) => properties?.model?.group === group,
+        );
+        if (groupMetadata.length > 0) {
+          return Object.assign({}, ...groupMetadata.map((e) => e.properties));
+        }
+        return {};
+      };
```

Comment:
You need to merge all matching permutations into a single object.
First check least specific to most specific.
Match only group first then match group + kind and finally group + kind + version.